plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/IDialogPageProvider.java

```diff
@@ -28,4 +28,17 @@ public interface IDialogPageProvider {
     @Nullable
     IDialogPage[] getDialogPages(boolean extrasOnly, boolean forceCreate);
 
+    /**
+     * Pages what should be saved during creation, even if they were
+     * never opened
+     *
+     * @return array of IDialogPage[] to create at the initialisation,
+     *      null if no additional pages should be created.
+     */
+    @Nullable
+    default IDialogPage[] getRequiredDialogPages() {
```

Comment:
Method name is not descriptive enough. I couldn't understand what does it mean, until I read the comment. I'd propose something like `getPagesContainingSettings`. Maybe you can figure out better name, but don't left just as is, please. It's not clear what does Required means - required for what?